### DISCUSSION TITLE
Inquiry Regarding SANDI Data Processing and Model Parameters

### DISCUSSION THREAD

**Discussion by HopperLiu** [daducci/AMICO#194] [Q&A]
Hi,

I am currently working on processing SANDI data using ex vivo mouse brain samples. For preprocessing, I have utilized MRtrix for DWIDenoise and MRDEGIBBS. The acquisition scheme is as follows:

- 71 samples, 4 shells
- 7 @ b=0, 16 @ b=8000.0, 16 @ b=6000.0, 16 @ b=4000.0, 16 @ b=1000.0

The model parameters are set as:

-     dis=0.003dis​=0.003
-     Rs=np.linspace(1e−6,1.2e−5,12)Rs=np.linspace(1e−6,1.2e−5,12)
-     din=np.linspace(0.00025,0.003,3)din​=np.linspace(0.00025,0.003,3)
-     disos=np.linspace(0.00025,0.003,3)disos​=np.linspace(0.00025,0.003,3)
-     λ1=1e−6λ1​=1e−6
-     λ2=4.1e−3λ2​=4.1e−3

Currently, I have successfully processed all parameters except for the DeDe parameter. Could you please provide guidance on how to proceed with the DeDe parameter, or suggest any additional steps to take?

Thank you for your assistance.

Best regards,
Zhuoheng

**Comment by nightwnvol** [comment]
Hi @HopperLiu,
did you try following the [SANDI tutorial](https://github.com/daducci/AMICO/wiki/SANDI) on the wiki page? If you encounter any issues during the processing, please feel free to share your script and the complete output.

**Reply by HopperLiu** [reply]
Hi @nightwnvol ,

Yes, I have followed your instructions. And I used SANDI Matlab Toolbox from Marco before. There is some different to generate SANDI parameters, especially De and fextra.

Here is my script:

 import amico
import os
import numpy as np
amico.core.setup()
root_pth = "/home/wanglab2/Desktop/1/38"
ae = amico.Evaluation(root_pth,
                      output_path='/home/wanglab2/Desktop/1/38/SANDI')
ae.set_config('doDirectionalAverage', True)
ae.set_config('doDebiasSignal', True)
ae.set_config('DWI-SNR', 30.0)
directory_path = '/home/wanglab2/Desktop/1/64'
delta = 0.0169 # Time between pulses [s]
small_delta = 0.009 # Pulses duration in [s]
TE = 0.033 # Echo time if different from delta+small_delta [s] (optional)
amico.util.sandi2scheme(os.path.join(directory_path,'NODDI2.bvals'),
                        os.path.join(directory_path,'NODDI.bvecs'),
                        delta,
                        small_delta,
                        TE_data=TE,
                        schemeFilename=os.path.join(directory_path,'SANDI_scheme.txt'),
                        bStep=100)

ae.load_data(dwi_filename=os.path.join(directory_path,'d_dwi.nii.gz'),
             scheme_filename=os.path.join(directory_path,'SANDI_scheme.txt'),
             mask_filename=os.path.join(directory_path,'mask.nii.gz'),
             b0_thr=50)

ae.set_model('SANDI')
d_is = 3e-3 # Intra-soma diffusivity [mm^2/s] (2 -> 3 37C) 2-3.2
Rs = np.array([1.55555556e-06, 3.44444444e-06, 4.44444444e-06, 5.33333333e-06, 6.00000000e-06, 6.55555556e-06, 8.11111111e-06, 9.55555556e-06, 1.19666667e-05]) # Radii of the soma [meters]
d_in = np.array([0.00091667, 0.00169444, 0.003]) # Intra-neurite diffusivitie(s) [mm^2/s]
d_isos = np.array([0.00036111, 0.00163889, 0.003]) # Extra-cellular isotropic mean diffusivitie(s) [mm^2/s]
ae.model.set(d_is, Rs, d_in, d_isos)
ae.set_config('doComputeRMSE', True)
ae.set_config('doComputeNRMSE', True)
ae.generate_kernels(regenerate=True, ndirs=2000)

ae.load_kernels()
lambda1 = 1e-6 # L1 regularization term (MUST be varied according to data)
lambda2 = 5e-3 # L2 regularization term (MUST be varied according to data)
ae.set_solver(lambda1=lambda1, lambda2=lambda2)
ae.fit()
ae.save_results(save_dir_avg=True)
ae.RESULTS()

DE:
![1](https://github.com/user-attachments/assets/92926302-d953-4e77-a0a4-4c7cfff826b6)

DIN:
![7](https://github.com/user-attachments/assets/5a350a59-5f60-412a-a5b7-1fbacf306436)

fextra:
![2](https://github.com/user-attachments/assets/94930509-1341-4e84-a86d-e6b28d989013)

fnuerite:
![3](https://github.com/user-attachments/assets/dedadde3-11d3-4342-9a72-e3e141400737)

fsoma:
![4](https://github.com/user-attachments/assets/137b56c8-2037-4478-b032-0f03baf62a3a)

Rsoma:
![5](https://github.com/user-attachments/assets/92b39856-377e-4e61-a534-aa4a5a4377b8)

Thank you very much for your assistance.

Best,
Zhuoheng
